docker/fetch_normalization_data.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+
+"""
+Fetches normalization data.
+
+Usage:
+
+    python docker/fetch_normalization_data.py
+
+"""
+
+import argparse
+import datetime
+import os
+import sys
+
+import psycopg2
+import requests
+
+
+# NOTE(willkg): These platforms come from the -prod data.
+PLATFORMS = ['Mac OS X', 'Linux', 'Windows', 'Unknown']
+
+
+def fetch_data_from_api(endpoint, product, version, platform, start_date, end_date):
+    """Fetches data from API"""
+    req = requests.get(endpoint, params={
+        'end_date': end_date,
+        'start_date': start_date,
+        'product': product,
+        'platforms': platform,
+        'versions': version,
+    })
+    if req.status_code != 200:
+        print('Error retrieving ADI: %s' % req.content)
+        return []
+
+    data = req.json()
+    return data['hits']
+
+
+def fetch_versions(conn, product_name):
+    """Fetch the version strings for a product from product_versions table"""
+
+    cursor = conn.cursor()
+    cursor.execute("""
+        SELECT version_string
+        FROM product_versions
+        WHERE
+            product_name=%s
+    """, (product_name,))
+    # NOTE(willkg): We don't want versions that end with "b" because they're weird.
+    versions = set([record[0] for record in cursor if not record[0].endswith('b')])
```

Comment:
It's a hot mess. 
https://github.com/mozilla-services/socorro/blob/d8080a5e45909fbe7d0ed415d49bd23e1794939c/socorro/external/postgresql/adi.py#L47-L58

Actually you don't need to do this at all. 
If you do and `/api/ADI` query for `?versions=55.0b&versions=55.0b3` you get the same output as if you'd done `?versions=55.0b3`. 
I.e. some ADI counts for `55.0b3` but `55.0b` omitted from the result. 

```
breakpad=> select version_string, adu_count, adu_date from product_adu a, product_versions pv where pv.product_version_id=a.product_version_id and version_string like '55.%' and version_string like '%b';
 version_string | adu_count | adu_date
----------------+-----------+----------
(0 rows)
```

So in conclusion, I suggest you simply change it to:
```python
versions = [record[0] for record in cursor]
```